source/dependence.c

```diff
@@ -2339,3 +2314,87 @@ void candl_compute_last_writer(osl_dependence_p dep, osl_scop_p scop) {
     dep = dep->next;
   }
 }
+
+void candl_dependence_remap(osl_dependence_p dependence,
+                            osl_scop_p union_scop,
+                            candl_label_mapping_p mapping) {
+  osl_relation_list_p rlist;
+  int i;
+
+  for ( ; dependence != NULL; dependence = dependence->next) {
+    dependence->label_source =
+      candl_label_mapping_find_original(mapping, dependence->label_source);
+    dependence->label_target =
+      candl_label_mapping_find_original(mapping, dependence->label_target);
+    
+    dependence->stmt_source_ptr = 
+      candl_statement_find_label(union_scop->statement,
+                                 dependence->label_source);
+    dependence->stmt_target_ptr = 
+      candl_statement_find_label(union_scop->statement,
+                                 dependence->label_target);
+    if (!dependence->stmt_source_ptr || !dependence->stmt_target_ptr)
+      continue;
+
+    for (i = 0, rlist = dependence->stmt_source_ptr->access;
+         rlist != NULL;
+         rlist = rlist->next, i++) {
+      if (i == dependence->ref_source)
+        dependence->ref_source_access_ptr = rlist->elt;
+      if (i == dependence->ref_target)
+        dependence->ref_target_access_ptr = rlist->elt;
+    }
+  }
+}
+
+/**
+ * \brief build the dependence graph of a scop.
+ * Contrary to the ::candl_dependence_single, scops should not have the usr data
+ * structure initialized in advannce.
+ * \param [in,out] scop     The scop to analyze.
+ * \param [in]     options  Analysis options.
+ * \returns                 A linked list of dependences, \c NULL if empty.
+ */
+osl_dependence_p candl_dependence(osl_scop_p scop,
+                                 candl_options_p options) {
+  osl_scop_p nounion_scop;
```

Comment:
I still think a "_" between "no" and "unions" is much clearer, even in the case of `scop_no_unions`.